Resources/config/orm.xml

```diff
@@ -102,6 +102,17 @@
             <factory class="%doctrine.orm.entity_manager.class%" method="create" />
         </service>
 
+        <service id="doctrine.orm.container_repository_factory" class="Doctrine\Bundle\DoctrineBundle\Repository\ContainerRepositoryFactory" public="false">
+            <argument type="service">
+                <service class="Symfony\Component\DependencyInjection\ServiceLocator">
+                    <tag name="container.service_locator" />
+                    <!-- services specified in compiler pass -->
```

Comment:
A better idea when using 3.3+ and creating a collector in a compiler pass is to use `\Symfony\Component\DependencyInjection\Compiler\ServiceLocatorTagPass::register`, which deduplicates service locators